### ISSUE TITLE
Missing properties not checked for function overloads

### ISSUE THREAD

**Issue by cherryblossom000** [microsoft/TypeScript#39739]
**TypeScript Version:**  3.9.7, 4.0.0-dev.20200724

**Search Terms:** overload function method property missing "not compatible" check

**Code**

```ts
function f(): {x: number}
function f() {
  return {}
}
```

**Expected behavior:**

TS should report that `x` is missing:

```
file.ts:3:3 - error TS2741: Property 'x' is missing in type '{}' but required in type '{ x: number; }'.

3   return {}
    ~~~~~~~~~

  file.ts:1:16
    1 function f(): {x: number}
                     ~
    'x' is declared here.


Found 1 error.
```

or that the overload is incompatible with its implementation:

```
file.ts:1:10 - error TS2394: This overload signature is not compatible with its implementation signature.

1 function f(): {x: number}
           ~

  file.ts:2:10
    2 function f() {
               ~
    The implementation signature is declared here.


Found 1 error.
```

**Actual behavior:** No error.

**Playground Link:** [link](https://www.typescriptlang.org/play/?ts=4.0.0-beta#code/GYVwdgxgLglg9mABMAFASgFyIN6IB5ZggC2ARgKYBOiAvgFCiSwLLo52KeKXlQiVJs9ekA)

**Related Issues:** I couldn't find any related issues.

**Comment by RyanCavanaugh** [comment]
This is the intended behavior since you're likely to have many different return types in each overload, and it'd be generally impossible to construct a value assignable to all of them simultaneously
